python/lance/__init__.py

```diff
@@ -13,3 +33,18 @@ def dataset(uri: str):
     """
     fmt = LanceFileFormat()
     return ds.dataset(uri, format=fmt)
+
+
+def write_table(table: pa.Table, destination: Union[str, Path], primary_key: str):
```

Comment:
So this requires holding everything in memory first right? If we have a bunch of images on S3, does this mean we need to hold them all in Arrow memory to convert to lance format?

Reply:
Right, so there will be a `StreamWriter` which basically opens a `DatasetWriter` and write batch records one by one.

It is another set of interfaces tho.

Similar to parquet https://arrow.apache.org/docs/cpp/parquet.html#writing-parquet-files